
### Tugas W2S2
42324001 - Oktova Samosir

# Soal 1 — Analisis Korelasi Data Cuaca

Pada soal ini dilakukan analisis korelasi pada dataset
`GlobalWeatherRepository.csv`.

Variabel yang dianalisis adalah:
- `temperature_fahrenheit`
- `pressure_mb`
- `air_quality_Sulphur_dioxide`

Korelasi digunakan untuk mengetahui hubungan linear antarvariabel.
Nilai korelasi berada pada rentang -1 sampai 1.



## Membaca Dataset

Dataset yang digunakan pada soal pertama adalah
`GlobalWeatherRepository.csv`.

In [39]:
import pandas as pd

weather_path = "GlobalWeather.csv"
weather_df = pd.read_csv(weather_path)

weather_df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


In [1]:
import pandas as pd
path = "shopping_trends.csv"
df = pd.read_csv(path)

df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,Cash,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,PayPal,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Credit Card,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,PayPal,Express,No,No,24,Venmo,Weekly


In [40]:
tabel_info_weather = pd.DataFrame({
    'No': range(1, len(weather_df.columns) + 1),
    'Column': weather_df.columns,
    'Non-Null Count': weather_df.notna().sum().values,
    'Null Count': weather_df.isna().sum().values,
    'Dtype': weather_df.dtypes.astype(str).values
})

tabel_info_weather

,No,Column,Non-Null Count,Null Count,Dtype
0,1,country,39594,0,str
1,2,location_name,39594,0,str
2,3,latitude,39594,0,float64
3,4,longitude,39594,0,float64
4,5,timezone,39594,0,str
5,6,last_updated_epoch,39594,0,int64
6,7,last_updated,39594,0,str
7,8,temperature_celsius,39594,0,float64
8,9,temperature_fahrenheit,39594,0,float64
9,10,condition_text,39594,0,str


# Analisis Customer Shopping Trends

Pada tugas ini, kita akan melakukan analisis terhadap dataset
Customer Shopping Trends.

Analisis yang dilakukan meliputi:
1. Analisis statistik pada kolom `Previous Purchases`.
2. Analisis korelasi `Age` dengan `Purchase Amount (USD)`.
3. Analisis korelasi `Purchase Amount (USD)` dengan `Review Rating`.

Library Pandas digunakan untuk membaca dan mengolah dataset.


## Variabel yang Digunakan

Sesuai dengan soal, analisis korelasi dilakukan terhadap tiga variabel,
yaitu `temperature_fahrenheit`, `pressure_mb`, dan
`air_quality_Sulphur_dioxide`.


In [41]:
tabel_variabel_weather = weather_df[
    [
        'temperature_fahrenheit',
        'pressure_mb',
        'air_quality_Sulphur_dioxide'
    ]
]

tabel_variabel_weather

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
0,79.8,1012,0.200
1,66.2,1012,0.100
2,73.4,1011,13.400
3,43.3,1007,0.200
4,78.8,1011,31.500
...,...,...,...
39589,61.2,1015,2.775
39590,70.2,1013,68.265
39591,71.9,1016,2.035
39592,96.8,1007,11.470


In [42]:
tabel_korelasi_weather = pd.DataFrame({
    'Variabel 1': [
        'temperature_fahrenheit',
        'temperature_fahrenheit',
        'pressure_mb'
    ],
    'Variabel 2': [
        'pressure_mb',
        'air_quality_Sulphur_dioxide',
        'air_quality_Sulphur_dioxide'
    ],
    'Nilai Korelasi': [
        weather_df['temperature_fahrenheit'].corr(
            weather_df['pressure_mb']
        ),
        weather_df['temperature_fahrenheit'].corr(
            weather_df['air_quality_Sulphur_dioxide']
        ),
        weather_df['pressure_mb'].corr(
            weather_df['air_quality_Sulphur_dioxide']
        )
    ]
})

tabel_korelasi_weather

,Variabel 1,Variabel 2,Nilai Korelasi
0,temperature_fahrenheit,pressure_mb,-0.486833
1,temperature_fahrenheit,air_quality_Sulphur_dioxide,-0.023152
2,pressure_mb,air_quality_Sulphur_dioxide,0.018620


## Interpretasi Hasil Korelasi

Berdasarkan hasil analisis korelasi, hubungan antara
`temperature_fahrenheit` dan `pressure_mb` memiliki nilai korelasi
sebesar **-0.486833**. Nilai tersebut menunjukkan hubungan negatif
dengan kekuatan sedang. Artinya, ketika `temperature_fahrenheit`
meningkat, `pressure_mb` cenderung menurun.

Hubungan antara `temperature_fahrenheit` dan
`air_quality_Sulphur_dioxide` memiliki nilai korelasi sebesar
**-0.021532**. Nilai tersebut mendekati 0 sehingga menunjukkan
hubungan negatif yang sangat lemah.

Hubungan antara `pressure_mb` dan
`air_quality_Sulphur_dioxide` memiliki nilai korelasi sebesar
**0.018620**. Nilai tersebut juga sangat dekat dengan 0 sehingga
menunjukkan hubungan positif yang sangat lemah.

Dari ketiga pasangan variabel tersebut, hubungan paling kuat
terdapat antara `temperature_fahrenheit` dan `pressure_mb`,
sedangkan hubungan paling lemah terdapat antara `pressure_mb`
dan `air_quality_Sulphur_dioxide`.

## Analisis Korelasi

Korelasi digunakan untuk mengetahui hubungan linear antara
variabel-variabel yang dianalisis.

Nilai korelasi berada pada rentang -1 sampai 1. Nilai positif
menunjukkan hubungan positif, nilai negatif menunjukkan hubungan
negatif, sedangkan nilai yang mendekati 0 menunjukkan hubungan
linear yang lemah.

In [31]:
df.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually


## Melihat Lima Data Pertama

Fungsi `head()` digunakan untuk melihat lima baris pertama dari
DataFrame.

Tahap ini dilakukan untuk mengetahui gambaran awal mengenai isi
dataset, seperti nama kolom dan contoh nilai yang terdapat di
dalamnya.

## Informasi Dataset

Fungsi `info()` digunakan untuk melihat informasi umum mengenai
dataset.

Informasi yang ditampilkan meliputi:
- jumlah baris dan kolom,
- nama setiap kolom,
- tipe data setiap kolom,
- jumlah data yang tidak kosong pada setiap kolom.

Informasi ini berguna untuk mengetahui struktur dataset sebelum
melakukan analisis lebih lanjut.

### 

In [32]:
stats_previous = df['Previous Purchases'].describe()

tabel_previous = pd.DataFrame({
    'Statistik': [
        'count',
        'mean',
        'std',
        'min',
        'Q1 (25%)',
        'Q2 (50%)',
        'Q3 (75%)',
        'max'
    ],
    'Nilai': [
        stats_previous['count'],
        stats_previous['mean'],
        stats_previous['std'],
        stats_previous['min'],
        stats_previous['25%'],
        stats_previous['50%'],
        stats_previous['75%'],
        stats_previous['max']
    ]
})

tabel_previous

,Statistik,Nilai
0,count,3900.000000
1,mean,25.351538
2,std,14.447125
3,min,1.000000
4,Q1 (25%),13.000000
5,Q2 (50%),25.000000
6,Q3 (75%),38.000000
7,max,50.000000


## Analisis Previous Purchases

Pada tahap ini, dilakukan analisis statistik deskriptif terhadap
kolom `Previous Purchases`.

Statistik yang dianalisis meliputi:
- Mean
- Standard deviation (std)
- Minimum (min)
- Maximum (max)
- Q1 atau kuartil pertama
- Q2 atau median
- Q3 atau kuartil ketiga

Fungsi `describe()` digunakan untuk memperoleh ringkasan statistik
dari kolom `Previous Purchases`.

In [33]:
stats_previous = df['Previous Purchases'].describe()

tabel_previous = pd.DataFrame({
    'Statistik': [
        'Mean',
        'Std',
        'Min',
        'Q1',
        'Q2 (Median)',
        'Q3',
        'Max'
    ],
    'Nilai': [
        stats_previous['mean'],
        stats_previous['std'],
        stats_previous['min'],
        stats_previous['25%'],
        stats_previous['50%'],
        stats_previous['75%'],
        stats_previous['max']
    ]
})

tabel_previous

,Statistik,Nilai
0,Mean,25.351538
1,Std,14.447125
2,Min,1.000000
3,Q1,13.000000
4,Q2 (Median),25.000000
5,Q3,38.000000
6,Max,50.000000


In [34]:
tabel_corr_age_purchase = pd.DataFrame({
    'Variabel 1': ['Age'],
    'Variabel 2': ['Purchase Amount (USD)'],
    'Nilai Korelasi': [corr_age_purchase]
})

tabel_corr_age_purchase

,Variabel 1,Variabel 2,Nilai Korelasi
0,Age,Purchase Amount (USD),-0.010424


## Interpretasi Korelasi Age vs Purchase Amount (USD)

Berdasarkan tabel, nilai korelasi antara `Age` dan
`Purchase Amount (USD)` adalah -0.010424.

Nilai tersebut bernilai negatif tetapi sangat dekat dengan 0.
Hal ini menunjukkan bahwa hubungan linear antara usia pelanggan
dan jumlah pembelian bersifat negatif dan sangat lemah.

Dengan demikian, berdasarkan dataset yang digunakan, `Age` tidak
menunjukkan hubungan linear yang kuat dengan `Purchase Amount (USD)`.

## Analisis Korelasi Purchase Amount (USD) vs Review Rating

Pada tahap ini dilakukan analisis korelasi antara
`Purchase Amount (USD)` dan `Review Rating`.

Analisis ini bertujuan untuk mengetahui arah dan kekuatan hubungan
linear antara jumlah pembelian dengan rating yang diberikan oleh
pelanggan.

In [35]:
corr_purchase_rating = df['Purchase Amount (USD)'].corr(
    df['Review Rating']
)

tabel_corr_purchase_rating = pd.DataFrame({
    'Variabel 1': ['Purchase Amount (USD)'],
    'Variabel 2': ['Review Rating'],
    'Nilai Korelasi': [corr_purchase_rating]
})

tabel_corr_purchase_rating

,Variabel 1,Variabel 2,Nilai Korelasi
0,Purchase Amount (USD),Review Rating,0.030776


## Interpretasi Korelasi Purchase Amount (USD) vs Review Rating

Berdasarkan tabel, nilai korelasi antara `Purchase Amount (USD)` dan
`Review Rating` adalah 0.030776.

Nilai tersebut bernilai positif tetapi sangat dekat dengan 0.
Hal ini menunjukkan bahwa hubungan linear antara jumlah pembelian
dan rating pelanggan bersifat positif dan sangat lemah.

Dengan demikian, berdasarkan dataset yang digunakan,
`Purchase Amount (USD)` tidak menunjukkan hubungan linear yang kuat
dengan `Review Rating`.

## Kesimpulan Customer Shopping Trends

Berdasarkan analisis yang telah dilakukan pada dataset
Customer Shopping Trends, diperoleh hasil sebagai berikut.

| Analisis | Hasil | Interpretasi |
|---|---:|---|
| Mean Previous Purchases | 25.351538 | Rata-rata pembelian sebelumnya sekitar 25.35 |
| Std Previous Purchases | 14.447125 | Data memiliki penyebaran sekitar 14.45 dari rata-rata |
| Min Previous Purchases | 1 | Pembelian sebelumnya paling sedikit adalah 1 |
| Q1 Previous Purchases | 13 | Sekitar 25% data berada pada atau di bawah 13 |
| Q2 / Median Previous Purchases | 25 | Nilai tengah data adalah 25 |
| Q3 Previous Purchases | 38 | Sekitar 75% data berada pada atau di bawah 38 |
| Max Previous Purchases | 50 | Pembelian sebelumnya paling banyak adalah 50 |
| Age vs Purchase Amount (USD) | -0.010424 | Hubungan linear negatif sangat lemah |
| Purchase Amount (USD) vs Review Rating | 0.030776 | Hubungan linear positif sangat lemah |

Berdasarkan hasil tersebut, kedua pasangan variabel yang dianalisis
memiliki nilai korelasi yang sangat dekat dengan 0. Hal ini
menunjukkan bahwa tidak terdapat hubungan linear yang kuat antara
`Age` dengan `Purchase Amount (USD)`, maupun antara
`Purchase Amount (USD)` dengan `Review Rating`.